src/server/pkg/sync/sync.go

```diff
@@ -0,0 +1,127 @@
+// Package sync provides utility functions similar to `git pull/push` for PFS
+package sync
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	pachclient "github.com/pachyderm/pachyderm/src/client"
+	"github.com/pachyderm/pachyderm/src/client/pfs"
+
+	protostream "go.pedge.io/proto/stream"
+	"golang.org/x/sync/errgroup"
+)
+
+// Pull clones an entire repo at a certain commit
+//
+// root is the local path you want to clone to
+// commit is the commit you want to clone
+// shard and diffMethod get passed to ListFile and GetFile. See documentations
+// for those functions for details on these arguments.
+func Pull(ctx context.Context, client pfs.APIClient, root string, commit *pfs.Commit, diffMethod *pfs.DiffMethod, shard *pfs.Shard) error {
```

Comment:
Do you think it'd make sense to expose this in `pachctl` so users can get data to and from local disk? Not as part of this PR though since that would expand to scope. Also I would have called this method `Checkout` that seems like the closest analogy in git to me but it's not perfect since git's checkout doesn't involve going over the network,